### PR TITLE
Fix WeakDynamicIndex and FactoryVec methods

### PR DESCRIPTION
**Pull request by gicrisf**

### CODE REVIEW

File: src/factory_advanced.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ The solution Relm4 chose was dynamic indices. These indices are updated automati
 {{#include ../examples/factory_advanced.rs:msg }}
 ```
 
-As you can see, we use a lot of `MsgIndex` aka `Weak<DynamicIndex>`. This allows us to always hold a reference to the dynamic index value.
+As you can see, we use a lot of `WeakDynamicIndex` aka `Weak<DynamicIndex>`. This allows us to always hold a reference to the dynamic index value.
```
Comment (AaronErhardt) [inline src/factory_advanced.md:33]: I think in this case we can remove the "aka `Weak<DynamicIndex>`" because it's its own type now.

**Review by AaronErhardt — CHANGES_REQUESTED** [review]
Thanks for the PR! There's just a small thing I'd like to adjust.

File: src/factory_advanced.md (hunk 1 above)
Comment (gicrisf) [reply src/factory_advanced.md:33]: > Thanks for the PR! There's just a small thing I'd like to adjust.

Really happy to give my support to the project, even if it's so small. Unfortunately, at this time I'm very busy: I promise to remove that part soon

**Comment by AaronErhardt** [comment]
Closing because this is outdated by now. Thanks for the effort regardless!

**Pull request closed without merging**